src/main/java/se/liu/ida/hefquin/engine/query/impl/SPARQLUnionPatternImpl.java

```diff
@@ -40,13 +43,15 @@ public void addSubPattern( final SPARQLGraphPattern p ) {
 
 	@Override
 	public boolean equals( final Object other ) {
+		
 		if ( this == other )
 			return true;
 
 		if ( other == null || !(other instanceof SPARQLUnionPattern) )
 			return false;
 
-		return ((SPARQLUnionPattern) other).getSubPatterns().equals(subPatterns);
+		return subPatterns.containsAll((Collection<?>) ((SPARQLUnionPattern) other).getSubPatterns());
```

Comment:
That's incorrect for two reasons:
First, it verifies only that every subpattern of the `other` union pattern is among the subpatterns of `this` union pattern; hence, it would return `true` even if `this` union pattern contains additional subpatterns that are not in the `other` union pattern.
Second, it does not consider the order of the subpatterns (`containsAll` checks only for containment not for order).

What was wrong with the original comparison?
```
return ((SPARQLUnionPattern) other).getSubPatterns().equals(subPatterns);
```